### ISSUE TITLE
APA score

### ISSUE THREAD

**Issue by BenxiaHu** [open2c/coolpuppy#128]
hello,
   thanks for developing the coolpuppy tool. it is very helpful. I have one question: what does APA score mean? would you like to explain it?
Best,
<img width="488" alt="image" src="https://user-images.githubusercontent.com/12387462/230942135-00b09462-6080-4ecc-8005-2c3b442d234a.png">

**Comment by Phlya** [comment]
Hello, by default it's the mean of the central 3x3 square of the pileup, which would typically be observed/expected enrichment, and indicative of average dot strength. Hope this helps.

**Comment by BenxiaHu** [comment]
thanks. 
when I ran coolpup.py

`coolpup.py  10000.mcool::/resolutions/10000  loop_coolpuppy.txt  --features_format bedpe  --expected 10000_expected.tsv -o 10kb_cooluppy`

do you think _observed/expected enrichment_ is **log(observed/expected)** value or not?

**Comment by Phlya** [comment]
The plots is probably displayed in log scale, but the values are not.

**Comment by BenxiaHu** [comment]
thanks. how to show log(observed/expected)? add this paramter" --scale log

**Comment by BenxiaHu** [comment]
https://user-images.githubusercontent.com/12387462/230942252-7dcbb530-7015-4568-9d99-5378e9221fb9.png

the scale in the right of the heatmap is too large (**0.083-12**). looks weird.

**Comment by Phlya** [comment]
The colour is log scaled, that's why it looks like this. In the CLI you can't do it, sorry, only using python API.

**Comment by BenxiaHu** [comment]
ImportError: cannot import name 'plotpup' from 'coolpuppy' (/python38/lib/python3.8/site-packages/coolpuppy/__init__.py)

I can import coolpuppy, but not from coolpuppy import plotpup.
do you know what causes such issue?

**Comment by Phlya** [comment]
Maybe you are using old coolpuppy?

**Comment by BenxiaHu** [comment]
yes, I just upgraded it (https://pypi.org/project/coolpuppy/).
but when I ran plotpup.py, I got the error:
AttributeError: 'DataFrame' object has no attribute 'iteritems'.

the previous version of coolpup.py did not show this error.

**Comment by Phlya** [comment]
Could you please post the full error?

**Comment by BenxiaHu** [comment]
plotpup.py --input_pups _significant_Short_10kb_cooluppy significant_Median_10kb_cooluppy \
>                         significant_Long_10kb_cooluppy --rownames Short Median Long --colnames test \
>            --output Neuron_significant_diff_10kb_cooluppy_mustache.pdf --scale log

ValueError: Multiple pileups for one of the conditions, ensure unique correspondence for each col/row combination

here is the new error.

**Comment by Phlya** [comment]
Yes, you need to specify what goes along rows vs columns. For example, I think you need `--rows features`.

**Comment by BenxiaHu** [comment]
thanks. that works. 
I have another question, given that I split the chromatin loops into 3 classes based on the loop length, do you think that I need to `--rescale` to rescale pileups to the same shape and size.
based on my knowledge, the long-range loops should show lower contact frequencey compared to short-range loops.
but my results look weird. I am not sure whether the coolpup.py causes this phenomenon
<img width="392" alt="image" src="https://user-images.githubusercontent.com/12387462/230986936-01bce465-9fe8-4b47-8d45-69cdcd042ad2.png">

**Comment by BenxiaHu** [comment]
is there any paramter to do fold enrichment of the center piexl over the nearby bins?

**Comment by BenxiaHu** [comment]
it works.
but the value is > 2. do you think the result makes sense?

**Comment by Phlya** [comment]
Well, without seeing the result, sounds good, makes sense. You have >2-fold enrichment of O/E interactions relative to background.

**Comment by BenxiaHu** [comment]
<img width="1285" alt="image" src="https://user-images.githubusercontent.com/12387462/231005701-72d88b95-1fa4-4826-8d36-e3644b6adca1.png">


here is the APA result. do you that makes sense?

**Comment by BenxiaHu** [comment]
but usually the fold enrichment is greater than 1 but less than 2.
